test/powershell/Language/Scripting/ActionPreference.Tests.ps1

```diff
@@ -207,6 +207,56 @@ Describe "Tests for (error, warning, etc) action preference" -Tags "CI" {
         $err.FullyQualifiedErrorId | Should -BeExactly "ParameterBindingFailed,Microsoft.PowerShell.Commands.TestPathCommand"
         $err.Exception.InnerException.InnerException | Should -BeOfType "System.Management.Automation.PSInvalidCastException"
     }
+
+    It 'Correctly interprets ErrorActionPreference on Write-Error when it''s reset in a finally block' {
+        try
+        {
+            $oldEAP = $ErrorActionPreference
+            $ErrorActionPreference = 'Stop'
+
+            try
+            {
+                Write-Error 'error'
+            }
+            finally
+            {
+                $ErrorActionPreference = $oldEAP
+            }
+        }
+        catch
```

Comment:
It would be great to add a command after `finally` block to check that the command works (and doesn't work in next test.)